### PR TITLE
Add Member.hasRole() and Member.hasRoles()

### PR DESCRIPTION
**Pull request by VixenKasai**
[contributing]: https://github.com/DV8FromTheWorld/JDA/wiki/5%29-Contributing

## Pull Request Etiquette

- [x] I have checked the PRs for upcoming features/bug fixes.
- [x] I have read the [contributing guidelines][contributing].

### Changes

- [x] Internal code
- [ ] Library interface (affecting end-user code) 
- [ ] Documentation
- [ ] Other: \_____ 

Closes Issue: NaN

## Description

Adds:
    -Member.hasRole(Role role) which will return a boolean based on if the member has the provided role
    -Member.hasRoles(Collection<Role> roles) which will return a boolean based on if the member has *all* provided roles

### CODE REVIEW

File: src/main/java/net/dv8tion/jda/internal/entities/MemberImpl.java [hunk 1]
```diff
@@ -182,6 +182,19 @@ public String getEffectiveName()
         return Collections.unmodifiableList(roleList);
     }
 
+    @Override
+    public boolean hasRole(Role role)
+    {
+        Checks.notNull(role, "Provided role");
```
Comment (Xirado) [inline src/main/java/net/dv8tion/jda/internal/entities/MemberImpl.java:188]:
```suggestion
        Checks.notNull(role, "Role");
```

File: src/main/java/net/dv8tion/jda/internal/entities/MemberImpl.java [hunk 2]
```diff
@@ -182,6 +182,19 @@ public String getEffectiveName()
...
+        return getRoles().stream().anyMatch(it -> it.getId().equals(role.getId()));
```
Comment (Xirado) [inline src/main/java/net/dv8tion/jda/internal/entities/MemberImpl.java:189]:
```suggestion
        return getRoles().stream().anyMatch(it -> it.equals(role));
```

**Review by Xirado — CHANGES_REQUESTED** [review]

File: src/main/java/net/dv8tion/jda/internal/entities/MemberImpl.java [hunk 3]
```diff
@@ -182,6 +182,19 @@ public String getEffectiveName()
...
+    }
+
+    @Override
+    public boolean hasRoles(Collection<Role> roles)
+    {
+        return getRoles().containsAll(roles);
+    }
```
Comment (Xirado) [inline src/main/java/net/dv8tion/jda/internal/entities/MemberImpl.java:196]: You should also check if the collection itself is null or elements in it are

**Comment by caneleex** [comment]
https://canary.discord.com/channels/125227483518861312/541288051243089920/561233859987898389

**Comment by MinnDevelopment** [comment]
You should just use `getRoles().contains(role)`.

**Pull request closed without merging**